src/txkube/test/test_network.py

```diff
@@ -421,7 +420,8 @@ def f1():
 
 
 @attr.s
-class Redirectable(proxyForInterface(IReactorSSL)):
+@implementer(IReactorSSL, IReactorTCP, IReactorTime)
```

Comment:
It's also possible to multiply inherit from `proxyForInterface` (last I checked, anyway).  eg:

    class Redirectable(proxyForInterface(IReactorSSL), proxyForInterface(IReactorTCP), proxyForInterface(IReactorTime)):
        ...

This avoids the need to define `seconds` and it is probably also less prone to future breakage because it will proxy the whole interface and not just a few hand-picked methods.

It would probably be nice if `proxyForInterface` accepted several interfaces so this could be:

    class Redirectable(proxyForInterface(IReactorSSL, IReactorTCP, IReactorTime)):
        ...

Since this fits on a single line better :wink: and avoids multiple inheritance.

Reply:
That sounds good to me.  I've updated the patch.

Reply:
I also added `IReactorPluggableNameResolver` because I was getting a DeprecationWarning:

```
/Users/crodrigues/twisted8/src/twisted/web/client.py:1496: DeprecationWarning: Passing HostnameEndpoint a reactor that does not provide IReactorPlug
```